v-next/core/src/types/providers.ts

```diff
@@ -0,0 +1,15 @@
+import type EventEmitter from "node:events";
+
+export interface RequestArguments {
+  readonly method: string;
+  readonly params?: readonly unknown[] | object;
+}
+
+export interface ProviderRpcError extends Error {
+  code: number;
+  data?: unknown;
+}
+
+export interface EIP1193Provider extends EventEmitter {
+  request(args: RequestArguments): Promise<unknown>;
+}
```

Comment:
We should define the interface that also includes the backwards compatibility methods

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5580/commits/7e133bda7d4aa5894222258027ffc72ead77ab29)